src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -1,14 +1,53 @@
 package core.basesyntax.service;
 
-import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
-    private final StorageDao storageDao = new StorageDaoImpl();
+    private static final int DEFAULT_AGE = 18;
+    private static final int MIN_PASSWORD_LENGTH = 6;
+    private final StorageDaoImpl people = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        checkNullUser(user);
+        checkValidPassword(user);
+        checkValidAge(user);
+        checkExistLogin(user);
+        return people.add(user);
+    }
+
+    private void checkNullUser(User user) {
+        if (user == null) {
+            throw new RegistrationException("User can't be null");
+        }
+        if (user.getLogin() == null) {
+            throw new RegistrationException("Login can't be null");
+        }
+        if (user.getAge() == null) {
+            throw new RegistrationException("Age can't be null");
+        }
+        if (user.getPassword() == null) {
+            throw new RegistrationException("Password can't be null");
+        }
```

Comment:
better to move null checks to their exclusive methods, so that `checkPassword` for example would check if it is not null and valid from the requirements point of view so that this method would be solely responsible for password validation.
Same for the other propertioes